enable markdown linter for onos-docs

adibrastegarnia opened the pull request:
To address issue #65 . We should merge the PRs related to the markdown linter in other repos first. 
We enabled the following rules: 

```
rule "MD001" # header levels increment by 1
rule "MD003", :style => :atx # header style - atx
rule "MD004", :style => :consistent # unordered list style - consistent
rule "MD005" # don't allow inconsistent indentation for list items
rule "MD006" # start bullets at beginning of line
rule "MD011" # Reversed link syntax
rule "MD014" # Dollar signs used before commands without showing output
rule "MD018" # No space after hash on atx style header
rule "MD023" # Headers must start at the beginning of the line
rule "MD024" # no multiple headers with same content
rule "MD028" # no blank lines within blockquote
rule "MD034" # Bare URL used
rule "MD037" # Spaces inside emphasis markers
rule "MD038" # Spaces inside code span elements
rule "MD039" # Spaces inside link text
rule "MD040" # Fenced code blocks should have a language specified
rule "MD046" #  Code block style
```
For more information: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

SeanCondon commented on build/base/Dockerfile:
```diff
@@ -3,11 +3,12 @@ RUN apk add libc6-compat
 RUN apk add --no-cache git
 RUN apk add --no-cache bash
 RUN apk add --no-cache openssh
```
I think these could be combined in to one `RUN` like below - would reduce the number of build stages

adibrastegarnia replied:
Good catch. will be improved

SeanCondon commented on docs/check.Dockerfile:
```diff
@@ -0,0 +1,8 @@
+FROM onosproject/onos-docs-base:latest
```
Can't see why it needs another Docker image - couldn't it be executed on the build image, with just a different entrypoint?

adibrastegarnia replied:
It should be possible. I was thinking perhaps it would be better to make them separate in case we want to update just the checker with more stuff but you are right. For now it would be better to have one image.

adibrastegarnia commented:
@SeanCondon 
I addressed your comments. For now, I provided a set of paths to markdown linter until I write a script that filters them properly. Before merge this, we need to merge PRs in onos-topo, onos-ztp, onos-test.
